src/transport/transport_dg.cc

```diff
@@ -505,7 +505,7 @@ void TransportDG<Model>::update_solution()
     // assemble stiffness matrix
     if (stiffness_matrix[0] == NULL
             || data_->subset(FieldFlag::in_main_matrix).changed()
-            || Model::flux_changed)
+            || data_->flow_flux.changed())
```

Comment:
These various changed flags are problematic since they have to be switched on and off. We should probably introduce the state counters. There is a stub `system/lazy_dependency.py` for that.